src/agent/communicator/include/ihttp_socket.hpp

```diff
@@ -4,10 +4,13 @@
 #include <boost/beast.hpp>
 #include <boost/system/error_code.hpp>
 
+#include <chrono>
 #include <string>
 
 namespace http_client
 {
+    constexpr int TIMEOUT_DEFAULT = 2;
+
```

Comment:
It the timeout is not configurable, I would remove it altogether from here and the method's signatures.